### ISSUE TITLE
[Request]Steering to position

### ISSUE THREAD

**Issue by Jesusemora** [UPBGE/blender#600]
The Steering actuator has an option to follow a target object, but this option is useless because of how the objects are cloned when added and they share the same name.
There is a way to use pathfinding with python to go from a position to another, but this does not work with obstacle simulation. The only way to use obstacle simulation is with Steering actuator.
What i ask is for the option to use a position (and being able to change it from python) instead of a target object name, in the Steering actuator. Target object may not even be necesary, since we can use obj.worldPosition instead.
this will facilitate making games with dynamic environments and multiple friendly AIs, wich could be needed in FPS, RTS, RPG, and even platformers and racing games.

**Comment by BluePrintRandom** [comment]
do you know python?
navmesh.findpath() is very handy

**Comment by Jesusemora** [comment]
> do you know python? navmesh.findpath() is very handy

> There is a way to use pathfinding with python to go from a position to another, but this does not work with obstacle simulation. The only way to use obstacle simulation is with Steering actuator.

navmesh.findpath(), as i said, does not work with obstacle simulation. it only works with a static navmesh, it does not avoid objects that are not in the navmesh, objects that are moving or are added, wich steering does.

**Comment by youle31** [comment]
actually, I think what you want is possible: http://pasteall.org/blend/index.php?id=47925 but tricky
(it seems there is a bug with sensor.status)

**Comment by Jesusemora** [comment]
i looked into the code in source/gameengine/ketji/KX_steeringactuator.cpp
m_target is the target object in the actuator, this is all the actuator does with it:
`const MT_Vector3& targpos = m_target->NodeGetWorldPosition();`
i think it gets the position. there are other lines of code, i think they are to check if m_target is an object:
`if (m_target)
		m_target->RegisterActuator(this);`
how hard would it be to give the actuator the position instead of an object? the only problem would be those who are already using it as it is. maybe we can put a toggle button to use position OR object?

**Comment by BluePrintRandom** [comment]
if type(target) is object:
    convert to vector
proceed assuming input is vector?

**Comment by Jesusemora** [comment]
> if type(target) is object: convert to vector proceed assuming input is vector?
yeah, something like that. many python functions already acept both an object or a "vec3" position, like navmesh.findpath(), obj.getVectTo(), or scene.addObject().
it would be a small change that would allow simpler scripts. an enemy can find a target with radar, then we get the objects from the sensor, get the position, and put it in the steering actuator. then we get more complex AI, that can attack allies, we can set decoys, we can do all kinds of things that are much harder with how it works now, and we avoid most of the bugs caused by cloned objects without extra code in the scripts, because the sensors do not pick objects by name, they pick the detected objects.
